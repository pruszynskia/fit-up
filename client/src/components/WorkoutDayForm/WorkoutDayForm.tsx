import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";

import useStyles from './WorkoutDayForm.styles';
import classnames from 'classnames';
import moment from "moment";
import {v4} from 'uuid'

import {
    Button,
    FormControl,
    InputLabel,
    MenuItem,
    Select,
    TextField
} from '@material-ui/core';
import { RootState, Workout, WorkoutDayDetails, WorkoutDayExerciseDetails } from "../../lib/types";
import { addWorkoutDayTest, editWorkoutDayTest } from "../../redux/actions";

interface WorkoutDayForm {
    data?: WorkoutDayDetails;
    date: moment.Moment;
    handleCloseF: Function
}

export default function WorkoutDayForm({ data, date, handleCloseF }: WorkoutDayForm) {

    const classes = useStyles();
    const dispatch = useDispatch();
    const edit = Boolean(data);
    const workouts = useSelector((state: RootState) => state.workout)

    const initialState: WorkoutDayDetails = {
        id: data?.id || v4(),
        workoutID:data?.workoutID || workouts[0]?.id,
        workoutName: data?.workoutName || workouts[0]?.name,
        date: data?.date || date.format("DDMMYYYY"),
        exercises: workouts[0]?.exercises.map((ex: any) => ({
            name: ex.name,
            bodyPart: ex.bodyPart,
            sets: 0,
            reps: 0,
            weight: 0,
            note: ""
        })) 
    }


    let [selectedName, setSelectedName] = useState<string>(initialState.workoutName)
    let [workoutDayFormData, setWorkoutDayFormData] = useState<WorkoutDayDetails>(data ? data : initialState);

    // Select
    const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
        setSelectedName(event.target.value as string);
        setWorkoutDayFormData({
            id: workouts.filter((w: Workout) => w.name === event.target.value as string)[0].id,
            workoutID: workouts.filter((w: Workout) => w.name === event.target.value as string)[0].id,
            workoutName: workouts.filter((w: Workout) => w.name === event.target.value as string)[0].name,
            date: date.format("DDMMYYYY"),
            exercises: workouts.filter((w: Workout) => w.name === event.target.value as string)[0].exercises.map((ex: any) => ({
                name: ex.name,
                bodyPart: ex.bodyPart,
                sets: 0,
                reps: 0,
                weight: 0,
                note: ""
            })) 
        })
    };

    // const handleSubmit = (e: any) => {
    //     e.preventDefault();

    //     if(edit) {
    //         dispatch({
    //             type: "EDIT_WORKOUT_DAY",
    //             payload: workoutDayFormData
    //         })
    //     } else {
    //         dispatch({
    //             type: "ADD_WORKOUT_DAY",
    //             payload: workoutDayFormData
    //         });
    //     };
    //     handleCloseF();
    // }
    
    const handleSubmit = (e: any) => {
        e.preventDefault();

        if(edit) {
            editWorkoutDayTest(dispatch, workoutDayFormData)
        } else {
            addWorkoutDayTest(dispatch, workoutDayFormData)
        }
        handleCloseF();
    };

    console.log("workouts", workouts[0]?.id)
    console.log("workoutDayFormData", workoutDayFormData)
    console.log("data", data)

    return (
        <div className={classes.root}>
            <span>{edit ? "EDIT WORKOUT" : "ADD WORKOUT"}</span>
            <form onSubmit={(e: any) => handleSubmit(e)}>  
                <div className={classnames(
                    classes.container,
                    classes.column
                    )}
                >
                    <span>{date.format("DD.MM.YYYY")}</span>
                    <FormControl>
                        <InputLabel>Workout</InputLabel>
                        <Select
                            defaultValue=""
                            onChange={handleChange}
                            value={selectedName}
                            >
                            {workouts.map((pos: Workout, id: any) => 
                                <MenuItem 
                                key={id} 
                                value={pos.name}
                                >
                                    {pos.name}
                                </MenuItem>
                            )}   
                        </Select>
                    </FormControl>
                    <div className={classnames(
                        classes.container,
                        classes.column,
                        )}
                        >
                        {Boolean(workoutDayFormData?.exercises?.length) && 
                            workoutDayFormData.exercises.map((ex: WorkoutDayExerciseDetails) =>
                            <div key={ex.name}
                            className={classnames(
                                classes.container,
                                classes.row
                                )}
                                >
                                    <span>{ex.name}</span>
                                    <TextField className={classes.offset}
                                        label="Weight"
                                        type="number"
                                        variant="outlined"
                                        value={ex.weight}
                                        onChange={(e: any) =>
                                            setWorkoutDayFormData({
                                                ...workoutDayFormData,
                                                exercises: workoutDayFormData.exercises.map((el:WorkoutDayExerciseDetails) => {
                                                    if(el.name === ex.name) {
                                                        return {
                                                            ...el,
                                                            weight: e.target.value
                                                        }
                                                    } else {
                                                        return {
                                                            ...el
                                                        }
                                                    }
                                                })
                                            })
                                        }
                                        />
                                    <TextField className={classes.offset}
                                        label="Reps"
                                        type="number"
                                        variant="outlined"
                                        value={ex.reps}
                                        onChange={(e: any) =>
                                            setWorkoutDayFormData({
                                                ...workoutDayFormData,
                                                exercises: workoutDayFormData.exercises.map((el:WorkoutDayExerciseDetails) => {
                                                    if(el.name === ex.name) {
                                                        return {
                                                            ...el,
                                                            reps: e.target.value
                                                        }
                                                    } else {
                                                        return {
                                                            ...el
                                                        }
                                                    }
                                                })
                                            })
                                        }
                                        />
                                    <TextField className={classes.offset}
                                        label="Sets"
                                        type="number"
                                        variant="outlined"
                                        value={ex.sets}
                                        onChange={(e: any) =>
                                            setWorkoutDayFormData({
                                                ...workoutDayFormData,
                                                exercises: workoutDayFormData.exercises.map((el:WorkoutDayExerciseDetails) => {
                                                    if(el.name === ex.name) {
                                                        return {
                                                            ...el,
                                                            sets: e.target.value
                                                        }
                                                    } else {
                                                        return {
                                                            ...el
                                                        }
                                                    }
                                                })
                                            })
                                        }
                                        />
                                </div>
                            )
                        }
                    </div>
                    <Button type="submit">
                        {edit ? "EDIT" : "ADD"}
                    </Button>
                </div>
            </form>
        </div>
    )
}