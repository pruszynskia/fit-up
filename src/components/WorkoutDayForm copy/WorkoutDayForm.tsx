import React, { useState, useEffect } from 'react';

import { useSelector, useDispatch } from "react-redux";

import useStyles from './WorkoutDayForm.styles';
import classnames from 'classnames';

import * as Moment from 'moment'
import {extendMoment} from 'moment-range'
const moment = extendMoment(Moment)
import {v4} from 'uuid'

import {
    Button,
    InputLabel,
    Select,
    MenuItem,
    FormControl,
    TextField,
} from '@material-ui/core';

import { WorkoutDayDetails, WorkoutDayExerciseDetails, Workout, RootState } from '../../lib/types'

interface WorkoutDayFormProps {
    handleCloseF: Function;
    data?: WorkoutDayDetails;
    date: moment.Moment

}

const WorkoutDayForm = ({ handleCloseF, data, date }: WorkoutDayFormProps) => {
    
    const edit = Boolean(data)
    const classes = useStyles();
    const dispatch = useDispatch();
    const workouts = useSelector((state: RootState) => state.workout)
    
    // Form
    const [wName, setWName] = React.useState<string>(data?.workoutName || workouts[0]?.name || '');

    const initialState: WorkoutDayDetails = {
        id: data?.id || v4(),
        workoutID: data?.workoutID || workouts[0].id,
        workoutName: data?.workoutName || workouts[0].name,
        date: data?.date || date.format("DDMMYYYY"),
        exercises: data?.exercises || workouts[0].exercises.map((ex: any) => ({
            name: ex.name,
            bodyPart: ex.bodyPart,
            sets: 0,
            reps: 0,
            weight: 0,
            note: ""
        }))
    }

    let [workoutDayFormData, setWorkoutDayFormData] = useState<WorkoutDayDetails>(data ? data : initialState);
    var selectedWorkout = initialState
 
    const handleSubmit = (e: any) => {
        e.preventDefault();

        if(edit) {
            dispatch({
                type: 'EDIT_WORKOUT_DAY',
                payload: workoutDayFormData
            })
        }
        else {
            dispatch({
                type: 'ADD_WORKOUT_DAY',
                payload: workoutDayFormData
            });
        }
        handleCloseF();
    };

     // Calendar
     var days: number[] = []
    
     if(days.length == 0)
         for(let i=1; i <= 31; i++) days.push(i)
    
    // Select
    const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
        setWName(event.target.value as string);
        setWorkoutDayFormData({...workoutDayFormData, workoutName: event.target.value as string})
    };

    return (
        <div className={classes.root}>
            <span>{edit ? "EDIT WORKOUT" : "ADD WORKOUT"}</span>
            <form onSubmit={(e: any) => handleSubmit(e)}>
                <div className={classnames(
                    classes.container,
                    classes.column
                    )}
                >
                    <span>
                        {date.format("DD.MM.YYYY")}
                    </span>
                    <FormControl>
                        <InputLabel>Workout</InputLabel>
                        <Select
                            id='id'
                            value={selectedWorkout.workoutName}
                            onChange={handleChange}    
                        >
                            {workouts.map((pos: Workout, id: any) => 
                                <MenuItem 
                                    key={id} 
                                    value={pos.name}
                                >
                                    {pos.name}
                                </MenuItem>
                                )}    
                        </Select>
                    </FormControl>
                    <div className={classnames(
                        classes.container,
                        classes.column,
                        )}
                    >
                        {Boolean(workoutDayFormData?.exercises?.length) && 
                                workoutDayFormData.exercises.map((ex: WorkoutDayExerciseDetails) => 
                            <div key={ex.name} 
                                className={classnames(
                                    classes.container,
                                    classes.row
                                )}
                            >
                                <span
                                    // onChange={() => 
                                    //     setWorkoutDayFormData({
                                    //         ...workoutDayFormData,
                                    //         workoutName: //workoutDayFormData.workoutName
                                    //     })
                                       
                                >{ex.name}</span>
                                <TextField className={classes.offset}
                                    label="Weight" 
                                    type="number"
                                    variant="outlined"
                                    value={ex.weight}
                                    onChange={(e: any) => 
                                        setWorkoutDayFormData({
                                            ...workoutDayFormData,
                                            exercises: workoutDayFormData.exercises.map((el: WorkoutDayExerciseDetails) => {
                                                
                                                if(el.name === ex.name){
                                                     return {
                                                        ...el,
                                                        weight: e.target.value
                                                    } 
                                                } else {
                                                    return {
                                                        ...ex
                                                    }
                                                }
                                            }),
                                        })
                                        }                                              
                                />
                                <TextField className={classes.offset}
                                    label="Reps" 
                                    type="number"
                                    variant="outlined"
                                    value={ex.reps}
                                    onChange={(e: any) => 
                                        setWorkoutDayFormData({
                                            ...workoutDayFormData,
                                            exercises: workoutDayFormData.exercises.map((el: WorkoutDayExerciseDetails) => {
                                                
                                                if(el.name === ex.name){
                                                     return {
                                                        ...el,
                                                        reps: e.target.value
                                                    } 
                                                } else {
                                                    return {
                                                        ...ex
                                                    }
                                                }
                                            }),
                                        })
                                        }         
                                />
                                <TextField className={classes.offset}
                                    label="Sets" 
                                    type="number"
                                    variant="outlined"
                                    value={ex.sets}
                                    onChange={(e: any) => 
                                        setWorkoutDayFormData({
                                            ...workoutDayFormData,
                                            exercises: workoutDayFormData.exercises.map((el: WorkoutDayExerciseDetails) => {
                                                
                                                if(el.name === ex.name){
                                                     return {
                                                        ...el,
                                                        sets: e.target.value
                                                    } 
                                                } else {
                                                    return {
                                                        ...ex
                                                    }
                                                }
                                            }),
                                        })
                                        }         
                                />
                            </div>
                        )}
                    </div>
                </div>
                <Button type="submit">
                    {edit ? "EDIT" : "ADD"}
                </Button>
            </form>
        </div>
    )
}

export default WorkoutDayForm;