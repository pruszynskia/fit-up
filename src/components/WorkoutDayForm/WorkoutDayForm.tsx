import React, { useState, useEffect } from 'react';

import { useSelector, useDispatch } from "react-redux";

import useStyles from './WorkoutDayForm.styles';
import classnames from 'classnames';

// import initialState from '../../temp/initialState.json'

import * as Moment from 'moment'
import {extendMoment} from 'moment-range'
const moment = extendMoment(Moment)
import { DatePicker, MuiPickersUtilsProvider } from "@material-ui/pickers";
import MomentUtils from '@date-io/moment';

import {
    Button,
    InputLabel,
    Select,
    MenuItem,
    FormControl,
    TextField,
} from '@material-ui/core';

import { WorkoutDayDetails, WorkoutDayExerciseDetails, Workout, RootState } from '../../lib/types'

interface WorkoutFormProps {
    handleClose: Function;
    data?: any;
}

const WorkoutDayForm = ({ handleClose, data}: WorkoutFormProps) => {
    const initialState: WorkoutDayDetails = {
        id: "",
        workoutID: "",
        workoutName: "",
        date: "",
        exercises: [{
            name: "",
            bodyPart: "",
            sets: 0,
            reps: 0,
            weight: 0,
            note: ""
        }]
    }

    const classes = useStyles();
    const dispatch = useDispatch();
    const workouts = useSelector((state: RootState) => state.workout)
    
    // Form
    const [selectedDate, handleDateChange] = useState<any>(new Date());
    const [wName, setWName] = React.useState<string>(workouts[0]?.name || '');
    var selectedWorkout = workouts.filter((w: Workout) => w.name === wName)[0]
    

    let [workoutDayFormData, setWorkoutDayFormData] = useState<WorkoutDayDetails>(data ? data : initialState);
 
    const handleSubmit = (e: any) => {
        e.preventDefault();

        dispatch({
            type: 'ADD_WORKOUT_DAY',
            payload: workoutDayFormData
        });
        handleClose();
    };

     // Calendar
     var days: number[] = []
    
     if(days.length == 0)
         for(let i=1; i <= 31; i++) days.push(i)
    

    // Select

    useEffect(() => {
     selectedWorkout = workouts.filter((w: Workout) => w.name === wName)[0]
    }, [wName])

    const handleChange = (event: React.ChangeEvent<{ value: unknown }>) => {
        setWName(event.target.value as string);
    };

    console.log("workoutDayFormData", workoutDayFormData)

    return (
        <div className={classes.root}>
            <form onSubmit={(e: any) => handleSubmit(e)}>
                <div className={classnames(
                    classes.container,
                    classes.column
                    )}
                >
                    <MuiPickersUtilsProvider utils={MomentUtils}>
                        <DatePicker
                            label="Date"
                            value={selectedDate}
                            onChange={handleDateChange}
                            animateYearScrolling
                        />
                    </MuiPickersUtilsProvider>
                    <FormControl>
                        <InputLabel>Workout</InputLabel>
                        <Select
                            id='id'
                            value={wName}
                            onChange={handleChange}    
                        >
                            {workouts.map((pos: Workout, id: any) => <MenuItem key={id} value={pos.name}>{pos.name}</MenuItem>)}    
                        </Select>
                    </FormControl>
                    <div className={classnames(
                        classes.container,
                        classes.column,
                        )}
                    >
                        {Boolean(selectedWorkout?.exercises?.length) && 
                                selectedWorkout.exercises.map((ex: WorkoutDayExerciseDetails) => 
                            <div key={ex.name} 
                                className={classnames(
                                    classes.container,
                                    classes.row
                                )}
                            >
                                <span>{ex.name}</span>
                                <TextField className={classes.offset}
                                    label="Weight" 
                                    type="number"
                                    variant="outlined"
                                    value={ex.weight}
                                    onChange={(e: any) => 
                                        setWorkoutDayFormData({
                                            ...workoutDayFormData,
                                            exercises: workoutDayFormData.exercises.map((el: WorkoutDayExerciseDetails) => (
                                                {
                                                    ...el,
                                                }
                                                )),
                                            
                                        })
                                        }                                              
                                    
                                />
                                <TextField className={classes.offset}
                                    label="Reps" 
                                    type="number"
                                    variant="outlined"
                                />
                                <TextField className={classes.offset}
                                    label="Sets" 
                                    type="number"
                                    variant="outlined"
                                />
                            </div>
                        )}
                    </div>
                                

                            </div>
                            <Button type="submit">
                                Add
                            </Button>
                        </form>
        </div>
    )
}

export default WorkoutDayForm;