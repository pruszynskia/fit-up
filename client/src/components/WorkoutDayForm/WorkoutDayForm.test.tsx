import React from "react";
import renderer from "react-test-renderer";
import moment from "moment";
import WorkoutDayForm from "./WorkoutDayForm";

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

interface WorkoutDayForm {
  date: moment.Moment;
  handleCloseF: () => void;
}

describe("Workout day form", () => {
  it("Should render as expected", () => {
    let date = moment();
    let handleCloseF = () => handleCloseF;

    const mockStore = configureStore()
    let store;

    const initialState = {
      workout: [{
        id: 1,
        workoutID: 1,
        workoutName: "name",
        date: "01012021",
        exercises: [{
          name: "exercise 1",
          bodyPart: "body part 1"
        }]
      }]
    }

    store = mockStore(initialState)
    const tree = renderer.create(
      <Provider store={store}>
      <WorkoutDayForm date={date} handleCloseF={handleCloseF} />
      </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
