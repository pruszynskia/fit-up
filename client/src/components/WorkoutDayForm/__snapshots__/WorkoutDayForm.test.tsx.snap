// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workout day form Should render as expected 1`] = `
<div
  className="makeStyles-root-1"
>
  <span>
    ADD WORKOUT
  </span>
  <form
    onSubmit={[Function]}
  >
    <div
      className="makeStyles-container-2 makeStyles-column-3"
    >
      <span>
        07.10.2021
      </span>
      <div
        className="MuiFormControl-root"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
        >
          Workout
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
        >
          <div
            aria-haspopup="listbox"
            className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
            value=""
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSelect-icon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
      <div
        className="makeStyles-container-2 makeStyles-column-3"
      >
        <div
          className="makeStyles-container-2 makeStyles-row-4"
        >
          <span>
            exercise 1
          </span>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-offset-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
            >
              Weight
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={0}
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
                >
                  <span>
                    Weight
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-offset-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
            >
              Reps
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={0}
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
                >
                  <span>
                    Reps
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-offset-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-filled"
              data-shrink={true}
            >
              Sets
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={0}
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-6 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-8 PrivateNotchedOutline-legendNotched-9"
                >
                  <span>
                    Sets
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-text"
        disabled={false}
        onBlur={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        <span
          className="MuiButton-label"
        >
          ADD
        </span>
      </button>
    </div>
  </form>
</div>
`;
